server/implementation/src/main/java/io/smallrye/graphql/bootstrap/Bootstrap.java

```diff
@@ -339,6 +340,9 @@ private GraphQLFieldDefinition createGraphQLFieldDefinitionFromOperation(String
         if (config != null && config.isTracingEnabled()) {
             decorators.add(new OpenTracingDecorator());
         }
+        if (config != null && config.isValidationEnabled()) {
+            decorators.add(new ValidationDecorator());
```

Comment:
Maybe we should apply the decorator only if `operation.hasArguments()`? Otherwise we unnecessarily run validation that won't do anything

Reply:
done